src/index.ts

```diff
@@ -1,144 +1,110 @@
-import Discord, { Client, Intents } from 'discord.js';
-const client = new Discord.Client({intents: [Intents.FLAGS.GUILDS, Intents.FLAGS.GUILD_MESSAGES, Intents.FLAGS.DIRECT_MESSAGES]});
+import Discord, { ApplicationCommandData, Client, Collection, Intents, Message, Snowflake, SnowflakeUtil } from 'discord.js';
+const client = new Discord.Client({ intents: [Intents.FLAGS.GUILDS, Intents.FLAGS.GUILD_MESSAGES, Intents.FLAGS.DIRECT_MESSAGES] });
 
 import dotenv from 'dotenv';
+import { cmdEvent, msgEvent } from './shared/interfaces'
 
 dotenv.config()
 
-// const DevIDs = ["170960451704717312"/*dislocated*/, "428934780764160010"/*tomkettle*/, "213389120829915136"/*tim*/]
-// const DevIDs:string[] = JSON.parse(process.env.ADMIN_DISCORD_IDS!); //Exclamation mark tells ts this var will never be null https://stackoverflow.com/a/57062363/15243027
-const DevIDs:string[] = process.env.DISCORD_ADMIN_IDS?.split(", ")!; //Exclamation mark tells ts this var will never be null https://stackoverflow.com/a/57062363/15243027
-
-const messageHandlers: {name: string; match: (msg: Discord.Message) => boolean; execute: (msg: Discord.Message) => void}[] = [
-    {
-        "name": "nice autoreply", //name for our reference only
-        "match": niceMatch,
-        "execute": niceExecute
-    },
-    {
-        "name": "woob autoreply",
-        "match": woobMatch,
-        "execute": woobExecute
-    }
-];
-
-const slashCommandHandlers: {command: string; execute: (interaction: Discord.CommandInteraction) => void}[] = [
-    {
-        "command": "ping",
-        "execute": pingExecute
-    },
-    {
-        "command": "admintest",
-        "execute": admintestExecute
+export const DevIDs: string[] = process.env.DISCORD_ADMIN_IDS?.split(", ")!;
+const Dev: boolean = process.env.NODE_ENV === "dev"
+
+const messageReplies: Collection<(msg: Message) => boolean, (event: msgEvent) => any> = new Collection();
+const slashCommands: Collection<ApplicationCommandData, (event: cmdEvent) => any> = new Collection();
+
+import fs from 'fs'
+
+fs.readdir('./dist/commands/message', (err, allFiles) => {
+    if (err) console.log(err);
+
+    let files = allFiles.filter(f => f.split('.').pop() === ('js')); // ignore .js.map files
+
+    if (files.length <= 0) console.log('No message replies found!');
```

Comment:
Could reword this cos not all of these will necessarily be for message replies.
(Getting nit picky here lmao)

Reply:
Wouldn't anything else be a slash command?

Reply:
```suggestion
    if (files.length <= 0) console.log('No message handlers found!');
```

Reply:
The hypothetical I can think of is something like if we wanted to count the occurrences of a word/emote etc.

Reply:
Like, I think this would be the best place for something like that to live